rel-eng/openshift-node.service

```diff
@@ -4,7 +4,7 @@ After=docker.service
 After=openshift-master.service
 After=openshift-sdn-master.service
 After=openshift-sdn-node.service
-Requires=docker.service
+Wants=docker.service
```

Comment:
Sorry if this is a repost, the threading looks weird in github so I suspect I screwed something up when replying the first time.

Currently, If you configure a node to use SDN it will permanently configure docker to use lbr0. The problem is that if you were to reboot lbr0 doesn't exist until openshift-node initializes it, so docker would fail, and openshift-node would never start and initialized the bridge. This allows docker to fail, openshift-node starts and the first thing it does is initialize the bridge and restart docker. It's still slightly messy, but thus far I've not seen it fail.

Long term I think we should fix this by reconfiguring the bridge only when bringing up the SDN. To do that we need to

    Have openshift-node write out an environment file to /run/openshift-sdn-ovs-docker-network (or some other ephemeral file that's removed on reboot) with DOCKER_NETWORK_OPTIONS
    Have openshift-sdn-ovs inject that as an optional EnvironmentFile via /usr/lib/systemd/system/docker.service.d/openshift-sdn-ovs.conf

A few examples of this pattern https://github.com/coreos/flannel/blob/master/dist/mk-docker-opts.sh
http://post-office.corp.redhat.com/archives/appinfra-list/2015-April/msg00688.html